src/search/randomdestinationairportpickingbycriteria.cpp

```diff
@@ -16,84 +16,87 @@
 *****************************************************************************/
 
 #include "search/randomdestinationairportpickingbycriteria.h"
+#include "search/randomdepartureairportpickingbycriteria.h"
 
 #include "geo/rect.h"
 
 #include <QRandomGenerator>
 
-int RandomDestinationAirportPickingByCriteria::randomLimit = 0;
+
 std::pair<int, atools::geo::Pos> *RandomDestinationAirportPickingByCriteria::data = nullptr;
 int RandomDestinationAirportPickingByCriteria::distanceMin = 0;
 int RandomDestinationAirportPickingByCriteria::distanceMax = 0;
-int RandomDestinationAirportPickingByCriteria::countResult = 0;
-bool RandomDestinationAirportPickingByCriteria::stopExecution = false;
+int RandomDestinationAirportPickingByCriteria::indexDeparture = -1;
 
-RandomDestinationAirportPickingByCriteria::RandomDestinationAirportPickingByCriteria(int indexDeparture)
+RandomDestinationAirportPickingByCriteria::RandomDestinationAirportPickingByCriteria(int threadId,
+                                                                                     int dataRangeIndexStart,
+                                                                                     int dataRangeLength)
 {
-  this->indexDeparture = indexDeparture;
+  this->threadId = threadId;
+  this->dataRangeIndexStart = dataRangeIndexStart;
+  this->dataRangeLength = dataRangeLength;
+  randomLimit = (dataRangeLength * 7) / 10;
+  this->offsettedData = RandomDestinationAirportPickingByCriteria::data + dataRangeIndexStart;
 }
 
-void RandomDestinationAirportPickingByCriteria::initStatics(std::pair<int, atools::geo::Pos> *data,
-                                                            int distanceMinMeter,
-                                                            int distanceMaxMeter,
-                                                            int countResult)
+void RandomDestinationAirportPickingByCriteria::initStatics(int distanceMinMeter,
+                                                            int distanceMaxMeter)
 {
-  RandomDestinationAirportPickingByCriteria::data = data;
   RandomDestinationAirportPickingByCriteria::distanceMin = distanceMinMeter;
   RandomDestinationAirportPickingByCriteria::distanceMax = distanceMaxMeter;
-  RandomDestinationAirportPickingByCriteria::countResult = countResult;
-  RandomDestinationAirportPickingByCriteria::randomLimit = (countResult * 7) / 10;
-  stopExecution = false;
+}
+
+void RandomDestinationAirportPickingByCriteria::initData(std::pair<int, atools::geo::Pos> *data,
+                                                         int indexDeparture)
+{
+  RandomDestinationAirportPickingByCriteria::data = data;
+  RandomDestinationAirportPickingByCriteria::indexDeparture = indexDeparture;
```

Comment:
indexDeparture must be set along data before each destination search thread batch is constructed